### DISCUSSION TITLE
Does watching a provider inside a hook makes the hook rebuild the widget ?

### DISCUSSION THREAD

**Discussion by felipe-xavier** [rrousselGit/flutter_hooks#363] [Q&A] [answered]
Say we have a custom hook that watches a provider and a Widget that uses the hook. Will a change in the state of the provider cause a rebuild of the Widget using the hook?

**A Widget:**
``` Dart
Widget build(BuildContext context, WidgetRef ref) {
  useCustomHook();
}
```

**useCustomHook:**
``` Dart
@override
void initHook() {
  final state = ref.watch(customProvider);
  ...
}
```

**Comment by rrousselGit** [comment, marked as answer]
Yes, the widget will rebuild if the provider changes

**Reply by csondi10** [reply]
How do we get `ref` in a hook?

**Reply by davidmartos96** [reply]
The recommended approach is to pass it as a parameter to the hook. There is no `useWidgetRef` hook
